### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
## Proposed changes

Fixes misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/mlx/actions/runs/7382182734/attempts/1#summary-20081642822

The action will report that the changes in this PR would make it happy: https://github.com/jsoref/mlx/actions/runs/7382182822/attempts/1#summary-20081643098

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [ ] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: benchmarks/python/blas/bench_gemv.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def get_gbyte_size(in_vec_len, out_vec_len, np_dtype):
     return float(N_iter_bench * N_iter_func * n_elem * item_size) / float(1024**3)
 
 
-def bench_with_in_len(ax, in_vec_len, out_vector_lens, dtype, tranpose):
+def bench_with_in_len(ax, in_vec_len, out_vector_lens, dtype, transpose):
```
Comment (jsoref) [inline benchmarks/python/blas/bench_gemv.py:136]: this is a guess...

File: docs/README.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ python -m http.server <port>
 
 and point your browser to `http://localhost:<port>`.
 
-### Push to Github Pages
+### Push to GitHub Pages
```
Comment (jsoref) [inline docs/README.md:29]: brand

File: docs/src/dev/extensions.rst [hunk 3]
```diff
@@ -485,7 +485,7 @@ each data type.
 
     instantiate_axpby(float32, float);
     instantiate_axpby(float16, half);
-    instantiate_axpby(bflot16, bfloat16_t);
+    instantiate_axpby(bfloat16, bfloat16_t);
```
Comment (jsoref) [inline docs/src/dev/extensions.rst:488]: ??

File: mlx/backend/metal/kernels/gemm/conv.h [hunk 4]
```diff
@@ -334,7 +334,7 @@ struct Conv2DBlockMMA {
       }
 
       simdgroup_barrier(mem_flags::mem_none);
-// Multiply and accumulate into resulr simdgroup matrices
+// Multiply and accumulate into result simdgroup matrices
```
Comment (jsoref) [inline mlx/backend/metal/kernels/gemm/conv.h:337]: ?

File: python/src/ops.cpp [hunk 5]
```diff
@@ -1742,7 +1742,7 @@ void init_ops(py::module_& m) {
       "a"_a,
       py::pos_only(),
       "source"_a,
-      "destiantion"_a,
+      "destination"_a,
```
Comment (jsoref) [inline python/src/ops.cpp:1745]: notable

**Comment by awni** [comment]
This is really awesome, thanks for doing that! I just checked the precommit and tests to be sure and everything looks good!

**Review by awni — APPROVED** [review]

**Pull request merged**